src/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -72,14 +72,16 @@ APPLICATION_INFO::GetOrCreateApplication(
         else
         {
             // another thread created the application
-            FINISHED(S_OK);   
+            FINISHED(S_OK);
         }
     }
 
     if (AppOfflineApplication::ShouldBeStarted(httpApplication))
     {
         LOG_INFO("Detected app_offline file, creating polling application");
         m_pApplication.reset(new AppOfflineApplication(httpApplication));
+        // Force status update
+        m_pApplication->QueryStatus();
```

Comment:
Why do you need to do this? Isn't the only way we get the status is via QueryStatus? If not, we should redesign this such that we don't need to call QueryStatus.